### ISSUE TITLE
Custom FieldArrayType losing state on add() or remove() v6

### ISSUE THREAD

**Issue by clark-stevenson** [ngx-formly/ngx-formly#3527]
**Question**

 I am upgrading from v5 to v6 of Formly and all has been amazing!  

However I am running into something which has me a bit miffed after several hours. 

I have a custom Component based on FieldArrayType.  It is quite complex to reproduce as it's for uploading files and relies on other layers of stuff but ultimately it comes down to `this.add` when a new file has been selected.   

Note that for testing, then `submit` just means `log`. 

When I select a file and then submit then I get the following expected output:

```
files: [
    {
         path: "https://lol.com/lol1.jpg", 
         success: false
    }
}
```

If I then go and upload an additional file then I get this unexpected output:

```
files: [
    {
        path: undefined,
        success: false,
    },
   {
       path: "https://lol.com/lol2.jpg",
       success: true,
   }
}
```

I have traced this down to the `this.add()` method is where the previous field becomes undefined but I can't for the file of me figure out why.  

I have a feeling I am somehow not updating the model correctly as this worked on v5:

```
     const index = this.field.fieldGroup.length;

     this.add(index, {
          path: undefined,
          success: false
     }}

     const newlyAdded = this.field.fieldGroup[index];
  
     // note this does a lot of stuff but I feel it's irrelevant because ultimately it will complete once the file is uploaded
     // and return the path
     const fileEntry = new FileUpload(blah);

     // once the file is uploaded, set the path
     fileEntry.onComplete.subscribe((path) => {
           const pathControl = newlyAdded.formControl.get("path");
           pathControl.setValue(path);
           
           const successControl = newlyAdded.formControl.get("success");
           successControl.setValue(true);
     });
```

But that doesn't explain why it works at all. 

Also!  The opposite is true with `this.remove()`.   

Let's say I go and select 3 files at once and then submit (instead of one at a time).  Then it's all good!

```
files: [
    {
         path: "https://lol.com/lol1.jpg",
         success: true,
    }, 
    {
         path: "https://lol.com/lol2.jpg",
         success: true,
    },
    {
         path: "https://lol.com/lol3.jpg",
         success: true,
    }
]
```

But now, if I `this.remove(index)` one of the files then everything becomes default.

```
files: [
    {
         path: undefined,
         success: false,
    },
    {
         path: undefined,
         success: false,
    }
]
```

Does anyone have any clue where I could investigate?  Thanks!

**Comment by aitboudad** [comment]
Could you please check if its related to the following issue https://github.com/ngx-formly/ngx-formly/issues/3455

**Comment by aitboudad** [comment]
Please feel free to leave a comment once you've a got a reproduction example so we can check it further on our side and provide a fixed if needed.

Thank you!

**Comment by aitboudad** [comment]
This issue has been fixed and released as part of [v6.1.7](https://github.com/ngx-formly/ngx-formly/releases/tag/v6.1.7) release.

Please let us know, in case you are still encountering a similar issue/problem.
Thank you!
